hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -3455,9 +3455,12 @@ private void validate() {
                 writeConcurrencyMode.name()));
       }
       if (writeConcurrencyMode == WriteConcurrencyMode.NON_BLOCKING_CONCURRENCY_CONTROL) {
+        //checkArgument(
+        // writeConfig.getTableType().equals(HoodieTableType.MERGE_ON_READ) && writeConfig.isSimpleBucketIndex(),
+        // "Non-blocking concurrency control requires the MOR table with simple bucket index");
         checkArgument(
-            writeConfig.getTableType().equals(HoodieTableType.MERGE_ON_READ) && writeConfig.isSimpleBucketIndex(),
-            "Non-blocking concurrency control requires the MOR table with simple bucket index");
+            writeConfig.getTableType().equals(HoodieTableType.MERGE_ON_READ),
```

Comment:
Only simple bucket index and MDT support NBCC, because MDT is actually with simple bucket index too, the loose of the check may cause confusions.